pkg/frontend/querymiddleware/limits.go

```diff
@@ -99,7 +105,8 @@ func (l limitsMiddleware) Do(ctx context.Context, r Request) (Response, error) {
 				"msg", "skipping the execution of the query because its time range is before the 'max query lookback' setting",
 				"reqStart", util.FormatTimeMillis(r.GetStart()),
 				"redEnd", util.FormatTimeMillis(r.GetEnd()),
-				"maxQueryLookback", maxQueryLookback)
+				"maxQueryLookback", maxQueryLookback,
+				"minBlockRetentionPeriod", blockRetentionPeriod)
```

Comment:
[nit] I think "min" could be misleading if compared with "maxQueryLookback" because for both of them we're actually taking the min value in the multi-tenant case. Given the multi-tenant case is not the common case, I would simplify it and just name it `blocksRetentionPeriod`.

```suggestion
				"blocksRetentionPeriod", blockRetentionPeriod)
```